src/main/java/seedu/address/model/time/TimeMask.java

```diff
@@ -101,6 +133,29 @@ private void occupySlots(int dayIndex, int startHourIndex, int endHourIndex) {
         weeklyOccupancy[dayIndex] = weeklyOccupancy[dayIndex] | mask;
     }
 
+    private void occupyMultipleDay(int startDay, long diff, int startHour, int endHour) {
+        int curr = startDay;
+        int counter = 1;
+
+        // find free time slots for the start date of the event
+        int startBits = Integer.parseInt("1".repeat(23 - startHour + 1), 2);
+        int mask = startBits << startHour;
+        weeklyOccupancy[curr] = weeklyOccupancy[curr] | mask;
+        curr = curr == 6 ? 0 : curr + 1;
+
+        // filled the in between days as busy for every time slots
+        while (counter < diff) {
+            startBits = Integer.parseInt("1".repeat(24), 2);
+            weeklyOccupancy[curr] = weeklyOccupancy[curr] | startBits;
+            curr = curr == 6 ? 0 : curr + 1;
```

Comment:
maybe can create a named constant in the class to use here instead of just the number '6'